R/teal_slices-store.R

```diff
@@ -1,44 +1,47 @@
-#' Store teal_slices object to a file
+#' Store `teal_slices` object to a file
 #'
 #' This function takes a `teal_slices` object and saves it to a file in `JSON` format.
 #' The `teal_slices` object contains information about filter states and can be used to
 #' create, modify, and delete filter states. The saved file can be later loaded using
 #' the `slices_restore` function.
 #'
-#' @param tss (`teal_slices`) object to be stored.
-#' @param file (`character(1)`) The file path where `teal_slices` object will be saved.
-#'  The file extension should be `".json"`.
-#'
-#' @details `Date` class is stored in `"ISO8601"` format (`YYYY-MM-DD`). `POSIX*t` classes are converted to a
+#' `Date` class is stored in `"ISO8601"` format (`YYYY-MM-DD`). `POSIX*t` classes are converted to a
 #' character by using `format.POSIX*t(usetz = TRUE, tz = "UTC")` (`YYYY-MM-DD {N}{N}:{N}{N}:{N}{N} UTC`, where
 #' `{N} = [0-9]` is a number and `UTC` is `Coordinated Universal Time` timezone short-code).
 #' This format is assumed during `slices_restore`. All `POSIX*t` objects in `selected` or `choices` fields of
 #' `teal_slice` objects are always printed in `UTC` timezone as well.
 #'
-#' @return `NULL`, invisibly.
+#' @param tss (`teal_slices`) object to be stored.
+#' @param file (`character(1)`) The file path where `teal_slices` object will be saved.
+#'  The file extension should be `".json"`.
 #'
-#' @keywords internal
+#' @return `NULL`, invisibly.
 #'
 #' @examples
+#' # use non-exported function from teal
+#' slices_store <- getFromNamespace("slices_store", "teal")
+#'
 #' # Create a teal_slices object
 #' tss <- teal_slices(
-#'   teal_slice(dataname = "data", varname = "var"),
-#'   teal_slice(dataname = "data", expr = "x > 0", id = "positive_x", title = "Positive x")
+#'   teal.slice::teal_slice(dataname = "data", varname = "var"),
+#'   teal.slice::teal_slice(dataname = "data", expr = "x > 0", id = "positive_x", title = "Positive x")
 #' )
 #'
 #' if (interactive()) {
 #'   # Store the teal_slices object to a file
 #'   slices_store(tss, "path/to/file.json")
```

Comment:
I think we should merge the `slices_restore` and `slices_store` documentation and add an example that creates a valid `json` file that the developer can inspect.

Or at least create it on the `tmp` directory as this example will create an error when copying & run

See #1072 as proposal